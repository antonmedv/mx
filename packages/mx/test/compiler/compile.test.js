const test = require('ava')
const compile = require('./compile')

test('compile single tag', t => {
  const code = `
  div
  `
  t.snapshot(compile(code))
})

test('compile html and components', t => {
  const code = `
  body
    Header title="hello world"
    Content
      section className="container"
    Footer value={value}
  `
  t.snapshot(compile(code))
})

test('compile class name from tag', t => {
  const code = `
  .class-name
  article.title.is-active
  `
  t.snapshot(compile(code))
})

test('compile text', t => {
  const code = `
  p hello world!
  p.
    here goes a long
    block of text
  div
    | Also a text
  `
  t.snapshot(compile(code))
})
