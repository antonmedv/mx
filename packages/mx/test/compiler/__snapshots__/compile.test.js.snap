// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compile class name from tag 1`] = `
"import React from \\"react\\"

export default function({}) {
  return React.createElement(\\"div\\", {
    className: \\"class-name\\",
  }), React.createElement(\\"article\\", { className: \\"title is-active\\" })
}
"
`;

exports[`compile conditions 1`] = `
"import React from \\"react\\"

export default function({}) {
  return foo == \\"bar\\"
    ? React.createElement(\\"p\\", null, \` Foo is bar\`)
    : React.createElement(\\"p\\", null, \` Nope\`)
}
"
`;

exports[`compile html and components 1`] = `
"import React from \\"react\\"

export default function({ value }) {
  return React.createElement(
    \\"body\\",
    null,
    React.createElement(Header, { title: \\"hello world\\" }),
    React.createElement(
      Content,
      null,
      React.createElement(\\"section\\", { className: \\"container\\" }),
    ),
    React.createElement(Footer, { value: value }),
  )
}
"
`;

exports[`compile single tag 1`] = `
"import React from \\"react\\"

export default function({}) {
  return React.createElement(\\"div\\", null)
}
"
`;

exports[`compile text 1`] = `
"import React from \\"react\\"

export default function({}) {
  return React.createElement(
    \\"p\\",
    null,
    \` hello world\`,
    \`!\`,
  ), React.createElement(
    \\"p\\",
    null,
    \`  here goes a long
  block of text
\`,
  ), React.createElement(
    \\"div\\",
    null,
    React.createElement(\\"text\\", null, \` Also a text\`),
  )
}
"
`;
